extensions/control-plane/transfer/transfer-pull-http-dynamic-receiver/src/test/java/org/eclipse/edc/connector/receiver/http/dynamic/TestFunctions.java

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.connector.receiver.http.dynamic;
+
+import org.eclipse.edc.connector.transfer.spi.types.DataRequest;
+import org.eclipse.edc.connector.transfer.spi.types.TransferProcess;
+import org.eclipse.edc.connector.transfer.spi.types.TransferProcessStates;
+import org.eclipse.edc.spi.types.domain.DataAddress;
+import org.eclipse.edc.spi.types.domain.transfer.DataFlowRequest;
+
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.eclipse.edc.connector.receiver.http.dynamic.HttpDynamicEndpointDataReferenceReceiver.HTTP_RECEIVER_ENDPOINT;
+
+public class TestFunctions {
+
+    public static TransferProcess createTransferProcess(String id, Map<String, String> properties) {
+        return TransferProcess.Builder.newInstance()
+                .id(id)
+                .state(TransferProcessStates.IN_PROGRESS.code())
+                .type(TransferProcess.Type.CONSUMER)
+                .dataRequest(DataRequest.Builder.newInstance()
+                        .destinationType("file")
```

Comment:
It looks confusing that we're testing an `HttpProxy` feature and the destination type here is `file`, that confirms that having the receiver endpoint in the destination address could have more sense (so this method would accept a `DataAddress` object)

Reply:
i agree that make more sense but as above it could be a leak of information since `dataDestination` is sent to the provider and unless we provide some encryption mechanism the provider will have the consumer token info in plain sight

Reply:
Yes, you're right, could the `DataAddress` be passed to this method? To make it more meaningful